tests/conftest.py

```diff
@@ -51,4 +54,49 @@ def get_info(data, filename):
         file_info = os.stat(data + '/' + filename)
         return file_info
     else:
-        assert 0  # file not found
+        assert filename is None  # Unable to download file as expected
+
+
+def start_server():
+    thread = stub_thread.stubThread(1, "TestThread")
+    thread.start()
+
+
+def stop_server():
+    for thread in threading.enumerate():
+        thread.join()
+
+
+def download_test(file_id, repo, file_names, sizes, conf, data):
+    runner = CliRunner()
+    try:
+        start_server()
+    except Exception as e:
```

Comment:
Also why are some catching exceptions and others not? If a test method throws an exception doesn't it fail anyways? Is that needed?